.travis.yml

```diff
@@ -205,16 +204,23 @@ after_success:
     git tag master;
     ./gradlew uploadArchives; 
   fi;
-
+- if [[ $TRAVIS_SECURE_ENV_VARS == true && $TRAVIS_EVENT_TYPE != cron ]]; then
```

Comment:
You duplicated all of this report uploading code in the after failure/after success block. Could that be condensed into the after script block and then the python script is run in the after failure block? I'm not sure what order these are all executed in...

Reply:
`after_script` happens **after** `after_failure` / `after_success` as far as I know so you can't do it nicely.

Reply:
I can set an environment variable here that tells it to run/notrun the python script later.  That's probably nicer.